src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/Pack.targets

```diff
@@ -99,7 +86,43 @@ Copyright (c) .NET Foundation. All rights reserved.
     <ConvertToAbsolutePath Condition="$(NuspecFile) != ''" Paths="$(NuspecFile)">
       <Output TaskParameter="AbsolutePaths" PropertyName="NuspecFileAbsolutePath" />
     </ConvertToAbsolutePath>
-    
+    <!--add a way to init all properties-->
+    <ItemGroup>
+      <!--This catches changes to properties-->
+      <NuGetPackInput Include="$(MSBuildAllProjects)"/>
+      <NuGetPackInput Include="@(_PackageFiles)"/>
+      <NuGetPackInput Include="@(_PackageFilesToExclude)"/>
+      <NuGetPackInput Include="@(_TargetPathsToAssemblies->'%(FinalOutputPath)')"/>
+      <NuGetPackInput Include="@(_TargetPathsToSymbols)"/>
+      <NuGetPackInput Include="@(_SourceFiles->Distinct())"/>
+      <NuGetPackOutput Include="$(RestoreOutputAbsolutePath)$(PackageId).$(PackageVersion).nuspec"/> 
```

Comment:
Nit: add a blank line before this to enhance readability (since this is a different list).